### PR TITLE
Add rfc  custom stage

### PR DESCRIPTION
**Pull request by kevin55156**
**What this PR does / why we need it**:
add RFC to implement custom stage

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipecd/issues/1089

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/rfcs/0010-custom-stage.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+This RFC introduces a new way to enable users to use “custom stages” that users defined in their pipelines.
+
+# Motivation
+
+Currently, users can use only stages that PipeCD have already defined. But some users want to define new stages by their use-cases as bellow. 
+
+- Deploying infrastructure by tools other than that PipeCD supports (terraform and kubernetes) such as SAM, cloud formation….
+- Building Docker images with Kaniko
+- Running integration tests
```
Comment (knanao) [inline docs/rfcs/0010-custom-stage.md:14]: These are the role of CD? 
I think such a function could be actualized, but I feel it's not appropriate mentioning it here.

File: docs/rfcs/0010-custom-stage.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+# Summary
+
+This RFC introduces a new way to enable users to use “custom stages” that users defined in their pipelines.
+
+# Motivation
+
+Currently, users can use only stages that PipeCD have already defined. But some users want to define new stages by their use-cases as bellow. 
+
+- Deploying infrastructure by tools other than that PipeCD supports (terraform and kubernetes) such as SAM, cloud formation….
+- Running End to End tests
```
Comment (khanhtc1202) [inline docs/rfcs/0010-custom-stage.md:13]: should not be listed here (CI roles)

File: docs/rfcs/0010-custom-stage.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+Note: If you install a binary in the above way, the name of the command includes the version like `sam-1.7.3 build`
+
+# Alternatives
+## How to define custom stages?
+1. Add bin executer to exisisting XXX_SYNC
+- Must implement for every app kind.
+- What app kind is SAM app?
+
+1. Add QUICK_SYNC stage
```
Comment (khanhtc1202) [inline docs/rfcs/0010-custom-stage.md:114]:
```suggestion
2. Add QUICK_SYNC stage
```

File: docs/rfcs/0010-custom-stage.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+- It's a stage with a lot of freedom, but it's inconsistent with the use cases being narrowed down.
+
+1. Add OtherApp to manage custom stages
```
Comment (khanhtc1202) [inline docs/rfcs/0010-custom-stage.md:117]:
```suggestion
3. Add OtherApp to manage custom stages
```

File: docs/rfcs/0010-custom-stage.md (hunk 2 above)
Comment (kevin55156) [reply docs/rfcs/0010-custom-stage.md:13]: I think this can be included CD roles because there is some cases where developers want to check pods run.

**Review by khanhtc1202 — APPROVED** [review]

**Pull request merged**